src/components/dashboard/browse/BeanItemDetailsContainer.js

```diff
@@ -27,17 +28,45 @@ class BeanItemDetailsContainer extends Component {
     update(page, reset, id) {
         const {dispatch} = this.props;
 
+        if (this.props.bean[id]) {
+            this.loadRoaster(page, reset, this.props.bean[id].roasterId);
+            return;
+        }
+
         dispatch(getItem(id)).then(() => {
-           const limit = 10;
-            let offset = (page - 1) * limit;
-            dispatch(getRoasterItems(this.props.bean.roasterId, reset ? 0 : offset, limit));
+            this.loadRoaster(page, reset, this.props.bean[id].roasterId);
         });
     }
 
+    bean() {
+        return this.props.bean[this.props.params.id];
```

Comment:
the singular `bean` is not actually singular in this case, right?  we should use `beans` for consistency imo, even if our intention is to only grab a singular bean